### PR TITLE
#25 migrate tools

### LINKED ISSUE #25: Migrate Tools (Module Tools, Simulation Tools, Chemprops, Easy Plotter)
Epic: #17

### PR DESCRIPTION
**Pull request by roryschadler**
feat (#25): Migrate Tools (Module Tools, Simulation Tools, Chemprops, Easy Plotter)

Migrated ToolSet pages for:
* Module Tools
    * Microstructure Characterization and Reconstruction Tools
        * Image Binarization Tools
        * Microstructure Characterization Tools
        * Microstructure Reconstruction Tools
* Simulation Tools

Migrated Tool pages for: 
* Dynamfit
* Dynamfit Result
* ChemProps
* ChemProps API key request
* CSV Easy Plotter

I did not migrate any tools under the Microstructure Characterization and Reconstruction Toolset, because this PR was already getting pretty big. I will write another ticket for those, and tackle it next. Both the migrated and as-yet unmigrated Tools will need to be revisited once the API is up and running, as they are reliant on it for creating and running their associated job, and then displaying the job's output. 

close #25

### CODE REVIEW

**Review by tholulomo — COMMENTED** [review]
@roryschadler can we set up a call to discuss this?

File: app/tests/unit/pages/nanomine/dynamfitresult.spec.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+global.console = {
+  log: jest.fn(), // console.log are ignored in tests
+
+  // Keep native behavior for other methods
+  error: console.error,
+  warn: console.warn,
+  info: console.info,
+  debug: console.debug
+}
+global.fetch = jest.fn()
```
Comment (tholulomo) [inline app/tests/unit/pages/nanomine/dynamfitresult.spec.js:14]: Can we make this re-usable if possible

**Review by tholulomo — APPROVED** [review]
We can address other issues with a TD as planned.

File: app/tests/unit/pages/nanomine/dynamfitresult.spec.js (hunk 1 above)
Comment (roryschadler) [reply app/tests/unit/pages/nanomine/dynamfitresult.spec.js:14]: Do we have a file for test configuration? It doesn't look like [test-setup.js](https://github.com/Duke-MatSci/materialsmine/blob/2f2edf54aa4b62b8b1203bd62294af9097367ea5/app/tests/jest/script/test-setup.js) is being called/used yet, should I add the global overrides to that and add it as a jest setup file?

File: app/src/modules/JobMgr.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+                  this.jobInputFiles[index].statusCode = v.status
+                  this.jobInputFiles[index].statusText = v.statusText
+                })
+                fetch(this.submitJobPath, {
+                  method: 'POST',
+                  jobId: this.jobId,
+                  jobType: this.jobType
+                })
+                  .then(function (res) {
+                    console.log('submit job success - statusCode: ' + res.status + ' statusText: ' + res.statusText)
```
[the remark this answers is not in the document]
Comment (roryschadler) [reply app/src/modules/JobMgr.js]: Resolved in 85f8c9a

File: app/src/components/nanomine/SmilesCanvas.vue [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+</script>
+
+<style scoped>
+  .smiles {
+  }
+  .canvas-wrapper {
+    padding: 0;
+    margin: 0;
+  }
+</style>
```
[the remark this answers is not in the document]
Comment (roryschadler) [reply app/src/components/nanomine/SmilesCanvas.vue]: We can! Resolved in 85f8c9a

File: app/src/components/nanomine/LineChart.vue [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+    // tooltipoOnMouseMove(that) {
+    //   // console.log("TOOLTIP ON MOVE: ", that);
+    //   const { x } = this.scales;
+    //   const x0 = x.invert(d3.mouse(that)[0]);
+    //   const i = this.bisectDate(this.data, x0, 1);
+    //   const d0 = this.data[i - 1];
+    //   const d1 = this.data[i];
+    // },
+    init () {
+      if (this.data && this.data.length > 0) {
```
[the remark this answers is not in the document]
Comment (roryschadler) [reply app/src/components/nanomine/LineChart.vue]: Resolved in 85f8c9a

File: app/src/components/nanomine/LineChart.vue [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+      required: true
+    },
+    // xlabel: {
+    //   type: String,
+    //   required: true
+    // },
+    // ylabel: {
+    //   type: String,
+    //   required: true
+    // },
```
[the remark this answers is not in the document]
Comment (roryschadler) [reply app/src/components/nanomine/LineChart.vue]: Resolved in 85f8c9a

File: app/src/components/nanomine/LineChart.vue [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+//  stroke-opacity: 1;
+//}
+
+//.grid path {
+//  stroke-width: 1;
+//}
+
+//.gf-area {
+//  fill: lightsteelblue;
+//}
```
[the remark this answers is not in the document]
Comment (roryschadler) [reply app/src/components/nanomine/LineChart.vue]: Resolved in 85f8c9a

File: app/src/components/nanomine/LineChart.vue [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+    //     });
+    // },
+    // tooltipoOnMouseMove(that) {
+    //   // console.log("TOOLTIP ON MOVE: ", that);
+    //   const { x } = this.scales;
+    //   const x0 = x.invert(d3.mouse(that)[0]);
+    //   const i = this.bisectDate(this.data, x0, 1);
+    //   const d0 = this.data[i - 1];
+    //   const d1 = this.data[i];
+    // },
```
[the remark this answers is not in the document]
Comment (roryschadler) [reply app/src/components/nanomine/LineChart.vue]: Resolved in 85f8c9a

File: app/src/components/nanomine/LineChart.vue [hunk 8]
```diff
@@ -0,0 +1,317 @@
...
+        <g class="focus">
+          <rect
+          width="100"
+          height="30"
+          class="tooltip"
+          y="0"
+          rx="4"
+          ry="4"
+          ></rect>
+        </g> -->
```
[the remark this answers is not in the document]
Comment (roryschadler) [reply app/src/components/nanomine/LineChart.vue]: Resolved in 85f8c9a

File: app/tests/unit/pages/nanomine/dynamfitresult.spec.js (hunk 1 above)
Comment (tholulomo) [reply app/tests/unit/pages/nanomine/dynamfitresult.spec.js:14]: Totally agree

Comment (roryschadler) [reply app/tests/unit/pages/nanomine/dynamfitresult.spec.js]: Great. Did that in 41e5d77. All current tests still run, but I'll bring up the change at the standup tomorrow just to let everyone know. Then I'll merge the PR.

**Pull request merged**